File: backend/app/schemas/health.py
from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    """Health status representation."""
    status: str = Field(default="healthy", json_schema_extra={"example": "healthy"})
    database: str = Field(default="connected", json_schema_extra={"example": "connected"})
    version: str = Field(default="1.0.0", json_schema_extra={"example": "1.0.0"})


class VersionResponse(BaseModel):
    """API version and environment metadata."""
    version: str = "1.0.0"
    name: str = "ContextVault API"
    environment: str = "development"

File: web-client/src/types/health.ts
/**
 * System health response from /api/health.
 */
export interface HealthResponse {
  status: 'healthy' | 'degraded' | 'unhealthy';
  database: 'connected' | 'disconnected';
  version: string;
}

/**
 * System version and environment response from /api/version.
 */
export interface VersionResponse {
  version: string;
  name: string;
  environment: string;
}

/**
 * Enhanced diagnostic info gathered by the client.
 */
export interface HealthDiagnostic {
  isOnline: boolean;
  status: 'healthy' | 'degraded' | 'unhealthy' | 'offline';
  databaseStatus: 'connected' | 'disconnected' | 'unknown';
  responseTimeMs: number;
  version: string;
  environment: string;
  targetUrl: string;
  lastCheckedAt: string;
  error?: string | null;
}
